tests/TestCase/Database/Driver/PostgresTest.php

```diff
@@ -171,4 +173,36 @@ public function testInsertReturning()
         $query = $translator($query);
         $this->assertEquals('FOO', $query->clause('epilog'));
     }
+
+    /**
+     * Test update with limit
+     *
+     * @return void
+     */
+    public function testUpdateLimit()
+    {
+        $driver = $this->getMockBuilder('Cake\Database\Driver\Postgres')
+            ->setMethods(['_connect', 'connection'])
+            ->setConstructorArgs([[]])
+            ->getMock();
+
+        $connection = $this->getMockBuilder('\Cake\Database\Connection')
+            ->setMethods(['connect', 'driver'])
+            ->setConstructorArgs([['log' => false]])
+            ->getMock();
+
+        $connection
+            ->expects($this->any())
+            ->method('driver')
+            ->will($this->returnValue($driver));
+
+        $query = new \Cake\Database\Query($connection);
+
+        $query->update('articles')
+            ->set(['title' => 'FooBar'])
+            ->where(['published' => true])
+            ->limit(5);
+
+        $this->assertEquals('UPDATE articles SET title = :c0 FROM (SELECT oid FROM articles WHERE published = :c1 LIMIT 5) __cake_update__ WHERE oid = (__cake_update__.oid)', $query->sql());
```

Comment:
I think this code is going to be removed. I'm going to merge in some changes from @lorenzo but the only way to be sure is to run tests again Postgres and not just verify the generated SQL.